providers/aws/components/ecs/setup.ftl

```diff
@@ -1124,129 +1124,43 @@
                                 schedule.Enabled
                     )]
 
-                    [#if networkMode == "awsvpc" ]
-
-                        [#-- Cloudfomation support not available for awsvpc network config which means that fargate isn't supported --]
-                        [#local eventRuleCliConfig =
-                            {
-                                "ScheduleExpression" : schedule.Expression,
-                                "State" : scheduleEnabled?then("ENABLED", "DISABLED")
-                            }]
-
-                        [#local eventTargetCliConfig =
-                            {
-                                "Targets" : [
-                                    {
-                                        "Id" : formatId(scheduleRuleId, "target"),
-                                        "Arn" : getExistingReference(ecsId, ARN_ATTRIBUTE_TYPE),
-                                        "EcsParameters" : {
-                                            "TaskCount" : schedule.TaskCount
-                                        } +
-                                        attributeIfTrue(
-                                            "NetworkConfiguration",
-                                            networkMode = "awsvpc",
-                                            {
-                                                "awsvpcConfiguration" : {
-                                                    "Subnets" : subnets,
-                                                    "AssignPublicIp" : publicRouteTable?then("ENABLE", "DISABLED")
-                                                }
-                                            }
-                                        ) +
-                                        attributeIfTrue(
-                                            "LaunchType",
-                                            engine == "fargate",
-                                            "FARGATE"
-                                        )
-                                    }
+                    [#if deploymentSubsetRequired("epilogue", false)]
+                        [@addToDefaultBashScriptOutput
+                            content=
+                                [
+                                    " case $\{STACK_OPERATION} in",
+                                    "   delete)",
+                                    "       # Manage Scheduled Event",
+                                    "       info \"Deleting Scheduled Task...\"",
+                                    "       delete_cloudwatch_event" +
+                                    "       \"" + region + "\" " +
+                                    "       \"" + scheduleRuleId + "\" " +
+                                    "       \"true\" || return $?",
+                                    "       ;;",
+                                    " esac"
```

Comment:
I think this needs to be a little bit different. To remove the existing cli created events we need to clean them up on the first deployment of the cloudformation based events, otherwise we run the risk of the scheduled task being executed twice or at random times. 

So I think we need to actually make this STACK_OPERATION `update|create` and add a pseduo stack output so that we can check and see if the cleanup has been completed. This makes sure that we don't accidentally delete the cloudwatch schedules created by cloudformation 

An example of this cleanup process is available in 
https://github.com/codeontap/gen3/blob/fece5bf524683e50f007aaf9982383d11b4c55d1/providers/aws/components/lb/setup.ftl#L465

Reply:
@roleyfoley I added schedules to resources and added epilogue script for clean up, hope it's correct now.